### PR TITLE
docs(release): update changelog for the 2024-02 release

### PR DESCRIPTION
**Pull request by dbolduc**
Output of `release/changes.sh` dropped verbatim in the first commit.

(Leave off `storagecontrol`, which was generated as experimental. We will say words about it when it goes GA.)

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13539)

### CODE REVIEW

**Review by alevenberg — APPROVED** [review]

**Comment by dbolduc** [comment]
> Did you want to wait to start the release until @scotthart finishes sample stuff?

I thought that was just #13530, but also I am not great at listening.

I will not merge this PR without @scotthart's approval.

**Review by scotthart — APPROVED** [review]
> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13539)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @dbolduc)

**Pull request merged**
